src/configure/mod.rs

```diff
@@ -509,14 +525,49 @@ async fn try_smtp_one_param(
 
     if let Err(err) = smtp.connect(context, param, addr, oauth2).await {
         info!(context, "failure: {}", err);
-        false
+        Err(ConfigurationError {
+            config: inf,
+            msg: err.to_string(),
+        })
     } else {
         info!(context, "success: {}", inf);
         smtp.disconnect().await;
-        true
+        Ok(())
     }
 }
 
+#[derive(Debug, thiserror::Error)]
+#[error("Trying {config}…\nError: {msg}")]
+pub struct ConfigurationError {
+    config: String,
+    msg: String,
+}
+
+async fn nicer_configuration_error(context: &Context, errors: Vec<ConfigurationError>) -> String {
+    let first_err = if let Some(f) = errors.first() {
+        f
+    } else {
+        return "".to_string();
+    };
+
+    if errors
+        .iter()
+        .all(|e| e.msg.to_lowercase().contains("could not resolve"))
```

Comment:
Is there any nicer way to do this than string comparison? Maybe I should experiment with downcasting

Reply:
You can move this check to the place where you store `err.to_string()` to `ConfigurationError` and match on the `err` instead. SMTP module has error type based on `thiserror` that can be matched. IMAP uses anyhow though, so need to change its return type first.

Are you sure showing original error to the user is not an option?

Reply:
> Are you sure showing original error to the user is not an option?

Well this would look like this:
![Screenshot_20200919-105727](https://user-images.githubusercontent.com/18012815/93663274-f27b5a80-fa66-11ea-926b-bad22c6cd194.png)
which is definitely better than the current state but does not look nice. OTOH even if users do not understand this, they will quickly guess that the missing internet is the problem.

If we say that this is not important anyway, we could also leave it as is (with string comparison) and be OK with the fact that it might break if the string is changed (and maybe even disable the concerning test).